src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -1052,8 +1051,19 @@ void CouplingSchemeConfiguration::checkIfDataIsExchanged(
       hasFound = true;
     }
   }
+
+  std::string dataName = "";
+  for (mesh::PtrMesh mesh : _meshConfig->meshes()) {
+    for (mesh::PtrData data : mesh->data()) {
+      if (data->getID() == dataID) {
+        dataName = data->getName();
+        break;
+      }
+    }
+  }
```

Comment:
Can you refactor this into a little helper function?
Maybe `meah::PtrData CouplingSchemeConfiguration::findDataByID(int ID);`
Return `nullptr` if there is no match. Then this becomes:

```
auto dataptr = findDataByName(dataName);
if (dataptr) {
  std::string dataName = dataptr->getName();
}
```

Reply:
But so far we only need this here. Isn't that a bit over-engineered?

Reply:
It's all about mental overhead at the call side. _(the inner loop should probably be a `findDataByID`in `Mesh`)_

Reply:
Done. See https://github.com/precice/precice/pull/826/commits/5413f41808606c87dc4e192ab30e63240e82284c.